ProfitApp/src/main/java/bsu/fpmi/profit/ProfitCRUD.java

```diff
@@ -0,0 +1,424 @@
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+        }
+    }
+
+    private PreparedStatement buildEditPreparedStatement( Ad ad, String id) throws SQLException {
+        if (ad == null || ad.getId() != null || ad.getVendor() != null || ad.getCreatedAt() != null) {
+            return null;
+        }
+        String sql = getEditSqlString(ad);
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        int i = 1;
+        if(ad.getLabel() != null && ad.getLabel().length() != 0){
+            preparedStatement.setString(i++, ad.getLabel());
+        }
+        if (ad.getDescription() != null && ad.getDescription().length() != 0) {
+            preparedStatement.setString(i++, ad.getDescription());
+        }
+        if (ad.getLink() != null) {
+            preparedStatement.setString(i++, ad.getLink());
+        }
+        if (ad.getPhotoLink() != null) {
+            preparedStatement.setString(i++, ad.getPhotoLink());
+        }
+        if (ad.getDiscount() != null) {
+            preparedStatement.setString(i++, ad.getDiscount());
+        }
+        if (ad.getValidUntil() != null) {
+            preparedStatement.setDate(i++, new java.sql.Date(ad.getValidUntil().getTime()));
+        }
+        preparedStatement.setInt(i, Integer.parseInt(id));
+        return  preparedStatement;
+    }
+
+    public boolean editOffer(String id, Ad ad){
+        try{
+            PreparedStatement preparedStatement = buildEditPreparedStatement(ad, id);
+            if(preparedStatement != null) {
+                preparedStatement.executeUpdate();
+                preparedStatement.close();
+
+                if (ad.getHashTags() != null) {
+                    editHashtags(id, ad);
+                }
+
+                return true;
+            }
+        }catch(SQLException e){
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public boolean deleteAd(String id){
+        try{
+            String sql = "delete from offer where offer_id = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, Integer.parseInt(id));
+            preparedStatement.executeUpdate();
+            return true;
+        }catch(SQLException e){
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private List<String> readTags(String id) throws SQLException {
+        List<String> tags = new ArrayList<>();
+
+        String sql = "select * from hashtags where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        while(rs.next()){
+            tags.add(rs.getString("tag"));
+        }
+        rs.close();
+        preparedStatement.close();
+        return tags;
+    }
+
+    private class ReviewsWrapper{
```

Comment:
Prefer `private static` classes when you don't need to manipulate fields of the parent class

Reply:
And move the class underneath the methods. Please always observe the sequence of class members: publics come before private-s, instance methods come before static members, etc.